main/src/ca/uwaterloo/flix/language/phase/Monomorpher.scala

```diff
@@ -562,143 +558,136 @@ object Monomorpher {
             // Types unify; use the substitution in the body.
             case Some(caseSubst) =>
               // Visit the base expression under the initial environment.
-              val e = visitExp(exp, env0, subst)
-              // Generate a fresh symbol for the let-bound variable.
+              val e = specializeExp(exp, env0, subst)
+              // Generate a fresh symbol.
```

Comment:
Loses information.

Reply:
its not a let-bound variable. I think these details are a bit redundant and as shown here are easily copied and not updated